src/vs/workbench/services/positronConsole/browser/positronConsoleService.ts

```diff
@@ -1921,90 +1931,142 @@ class PositronConsoleInstance extends Disposable implements IPositronConsoleInst
 	 * Processes pending input.
 	 */
 	private async processPendingInput(): Promise<void> {
-		// If there isn't a pending input runtime item, return.
-		if (!this._runtimeItemPendingInput) {
+		// If we are already processing pending input and the `await` below allowed us to loop
+		// back into here, refuse to process anything else right now.
+		if (this._pendingInputState !== 'Idle') {
 			return;
 		}
 
-		// Get the index of the pending input runtime item.
-		const index = this.runtimeItems.indexOf(this._runtimeItemPendingInput);
+		this._pendingInputState = 'Processing';
 
-		// This index should always be > -1, but be defensive.
-		if (index > -1) {
-			this._runtimeItems.splice(index, 1);
+		try {
+			return await this.processPendingInputImpl();
+		} finally {
+			this._pendingInputState = 'Idle';
 		}
```

Comment:
The way I see it:

```ts
// Does not do what we want, i.e. does not return the result of the impl helper
await this.processPendingInputImpl();

// Breaks the async / await abstraction by returning a promise instead of a value
return this.processPendingInputImpl();

// Matches the async /await abstraction and correctly returns the result
return await this.processPendingInputImpl();
```